src/Symfony/Component/Security/Http/Tests/Firewall/AccessListenerTest.php

```diff
@@ -12,198 +12,331 @@
+        ;
+
+        $this->authenticationManager
+            ->expects($this->once())
+            ->method('authenticate')
+            ->with($this->equalTo($token))
+            ->will($this->returnValue($token))
+        ;
+
+        $this->tokenStorage
+            ->expects($this->once())
+            ->method('setToken')
+            ->with($this->equalTo($token))
+        ;
+
+        $this->accessDecisionManager
+            ->expects($this->once())
+            ->method('decide')
+            ->with($this->equalTo($token), $this->equalTo(array('foo' => 'bar')), $this->equalTo($this->request))
+            ->will($this->returnValue(true))
+        ;
+
+        $listener = new AccessListener(
+            $this->tokenStorage,
+            $this->accessDecisionManager,
+            $this->accessMap,
+            $this->authenticationManager,
+            $authorizationChecker
+        );
+
+        $this->event
+            ->expects($this->any())
+            ->method('getRequest')
+            ->will($this->returnValue($this->request))
+        ;
+
+        $listener->handle($this->event);
+    }
+
+    public function testHandleWithAuthorizationCheckerWhenAlwaysAuthenticateIsDisabledOnAuthenticatedToken()
+    {
+        $token = new ConcreteToken('user', array('ROLE_FOO'));
+
+        $authorizationChecker = new AuthorizationChecker(
+            $this->tokenStorage,
+            $this->authenticationManager,
+            $this->accessDecisionManager,
+            false
+        );
+
+        $this->tokenStorage
+            ->expects($this->once())
+            ->method('getToken')
+            ->will($this->returnValue($token))
+        ;
+
+        $this->accessMap
+            ->expects($this->any())
+            ->method('getPatterns')
+            ->with($this->equalTo($this->request))
+            ->will($this->returnValue(array(array('foo' => 'bar'), null)))
+        ;
+
+        $this->authenticationManager
+            ->expects($this->never())
+            ->method('authenticate')
+        ;
+
+        $this->tokenStorage
+            ->expects($this->never())
+            ->method('setToken')
+        ;
+
+        $this->accessDecisionManager
+            ->expects($this->once())
+            ->method('decide')
+            ->with($this->equalTo($token), $this->equalTo(array('foo' => 'bar')), $this->equalTo($this->request))
+            ->will($this->returnValue(true))
+        ;
+
+        $listener = new AccessListener(
+            $this->tokenStorage,
+            $this->accessDecisionManager,
+            $this->accessMap,
+            $this->authenticationManager,
+            $authorizationChecker
         );
 
-        $event = $this->getMockBuilder('Symfony\Component\HttpKernel\Event\GetResponseEvent')->disableOriginalConstructor()->getMock();
+        $this->event
+            ->expects($this->any())
+            ->method('getRequest')
+            ->will($this->returnValue($this->request))
```

Comment:
Not sure if there's a standard defined for this, but you can also write:

```php
$this->event->method('getRequest')->willReturn($this->request);
```

Reply:
Personally I do prefer shortcuts like `willReturn`, but I decided to stick with style that was before